pyo3-derive-backend/src/pymethod.rs

```diff
@@ -12,8 +12,56 @@ pub fn gen_py_method(
 ) -> syn::Result<TokenStream> {
     check_generic(name, sig)?;
 
-    let doc = utils::get_doc(&meth_attrs, true);
-    let spec = FnSpec::parse(name, sig, meth_attrs)?;
+    let spec = FnSpec::parse(name, sig, &mut *meth_attrs)?;
+
+    let text_signature = match &spec.tp {
+        FnType::Fn | FnType::PySelf(_) | FnType::FnClass | FnType::FnStatic => {
+            utils::parse_text_signature_attrs(&mut *meth_attrs, name)?
+        }
+        FnType::FnNew => {
+            // try to parse anyway to give better error messages
+            if let Some(type_signature) = utils::parse_text_signature_attrs(
+                &mut *meth_attrs,
+                &syn::Ident::new("__new__", name.span()),
+            )? {
+                return Err(syn::Error::new_spanned(
+                    type_signature,
+                    "type_signature not allowed on __new__, put it on the struct definition instead",
```

Comment:
it means that instead of writing:
```rust
#[pyclass]
struct MyClass {
}

#[pymethods]
impl MyClass {
    #[new]
    #[text_signature = "(a, b, c)"]
    fn new(...) {...}
}
```
This should be written instead:
```rust
#[pyclass]
#[text_signature = "(a, b, c)"]
struct MyClass {
}

#[pymethods]
impl MyClass {
    #[new]
    fn new(...) {...}
}
```

Do you have any suggestions for how to rephrase the error message?

Reply:
```rust
#[pyclass]
#[text_signature = "(a, b, c)"]
struct MyClass {
}
```
So this `(a, b, c)` is a signature of `__init__`?
Then I suggest adding 'If you want to add a signature to __init___, '

Reply:
fixed, though I used `__new__` instead of `__init__` since PyO3 doesn't support `__init__`